lib/grape/router/pattern.rb

```diff
@@ -0,0 +1,55 @@
+require 'forwardable'
+require 'mustermann/rails'
+
+module Grape
+  class Router
+    class Pattern
+      DEFAULT_PATTERN_OPTIONS   = { uri_decode: true }.freeze
+      DEFAULT_SUPPORTED_CAPTURE = [:format, :version].freeze
+
+      attr_reader :origin, :path, :capture, :pattern
+
+      extend Forwardable
+      def_delegators :pattern, :named_captures, :params
+      def_delegators :@regexp, :===
+      alias_method :match?, :===
+
+      def initialize(pattern, options = {})
+        @origin  = pattern
+        @path    = build_path(pattern, options)
+        @capture = extract_capture(options)
+        @pattern = Mustermann::Rails.new(@path, pattern_options)
```

Comment:
Maybe we should contribute Mustermann::Grape? Do we really always want Rails semantics here?